macros/src/path.rs

```diff
@@ -0,0 +1,244 @@
+use quote::{format_ident, quote};
+use proc_macro2::TokenStream;
+use syn::{
+    parse::{Parse, ParseStream},
+    Error, Lit, Ident,LitStr, Token, Result,
+};
+
+
+#[derive(Clone,Debug)]
+pub enum CustomPath{
+    Str(String),
+    Static(syn::Path), 
+    Fn(syn::Path),  
+    Env(syn::LitStr),     
+}
+
+type FnOutputPathBody = ( TokenStream, Option<TokenStream> );
+
+impl CustomPath {
+    
+    pub fn get_path_and_some_decl(&self, ts_name: &String ) -> FnOutputPathBody {
+
+        match self {
+
+            Self::Str(input) => { Self::str_path(input,ts_name) },
+
+            Self::Static(input) => { Self::static_path(input,ts_name) },
+
+            Self::Fn(input) => { Self::fn_path(input,ts_name) },
+
+            Self::Env(input) => { Self::env_path(input,ts_name) },
+
+        } 
+    }
+
+    fn str_path( input: &String, ts_name: &String ) -> FnOutputPathBody {
+
+        let path = 
+        if input.ends_with('/') {
+            format!("{}{}.ts", input, ts_name)
+        }  else {
+            input.to_owned()
+        };
+
+        return (quote!(#path),None);
+    }
+
+    fn static_path( input: &syn::Path, ts_name: &String ) -> FnOutputPathBody {
+
+        let path_ident      = format_ident!("path");
+        let stat_path_ident = format_ident!("PATH");
+        let path_decl = quote! {
+
+            static #stat_path_ident: std::sync::OnceLock<String> = std::sync::OnceLock::new();
+        
+            let #path_ident = #stat_path_ident.get_or_init( ||
+                {
+                    if #input.ends_with('/')  {
+                        format!("{}{}.ts", #input, #ts_name)
+                    } else {
+                        format!("{}",#input)
+                    }
+                } 
+            );
+        };
+
+        ( quote!(#path_ident), Some(path_decl) )
+    }
+
+    fn fn_path( input: &syn::Path, ts_name: &String ) -> FnOutputPathBody { 
+        
+        ( quote!{#input (#ts_name)?}, None)
+    }
+
+    fn env_path( input: &LitStr, ts_name: &String ) -> FnOutputPathBody {
+
+        let path_ident = format_ident!("path");
+
+        let path_decl = quote!{
+
+            let #path_ident = if std::env!(#input).ends_with('/') {
+                std::concat!(std::env!(#input),#ts_name,".ts")
+            } else {
+                std::env!(#input)
+            };
+        };
+
+        ( quote!(#path_ident), Some(path_decl) )
+    }
+
+}
+
+impl Parse for CustomPath {
+
+    fn parse(input: ParseStream) -> Result<CustomPath> {
+        input.parse::<Token![=]>()?;
+        let span = input.span();
+
+        let msg = 
+"expected arguments for 'export_to':
+
+1) string literal 
+    #[ts(export_to = \"my/path\")] 
+
+2) static or constant variable name 
+
+    #[ts(export_to = MY_STATIC_PATH)]
+    #[ts(export_to = crate::MY_STATIC_PATH)]
+
+Note: This option is available for Rust 1.7.0 and higher!
```

Comment:
This feels like a typo, did you really mean 1.7.0?

Reply:
It's not a typo [std::sync::OnceLock](https://doc.rust-lang.org/stable/std/sync/struct.OnceLock.html)  is stabilised in Rust 1.70.0, probably available 1.63. nightly only.

The idiomatic way is to check the Rust version, but  [#[cfg(version(..))])](https://github.com/rust-lang/rfcs/pull/2523)  isn't stabilised as well.

The worst-case scenario is a compile-time error, informing the user that `std::sync::OnceLock` is only available in nightly.

Additionally, since `ts-rs` currently depends on `lazy_static`, there's a good chance that this dependency will be dropped in favour of the standard library types increasing the MSRV and making this issue less relevant over time.

not sure what to do with it it's up to you really.

Reply:
So you meant 1.70.0 right?

Reply:
of course 1.70 )))

Reply:
And the code says 1.7... you're missing a zero

Reply:
That's right, it is a typo )))

Reply:
Alright, in that case, we can just change the MSRV to 1.70.0 and we don't have to worry about it